pkg/scheduler/framework/runtime/metrics_recorder.go

```diff
@@ -42,20 +43,23 @@ type metricsRecorder struct {
 	// how often the recorder runs to flush the metrics.
 	interval time.Duration
 
-	// stopCh is used to stop the goroutine which periodically flushes metrics. It's currently only
+	// ctx and cancel are used to stop the goroutine which periodically flushes metrics. They're currently only
 	// used in tests.
-	stopCh chan struct{}
+	ctx    context.Context
+	cancel context.CancelFunc
```

Comment:
`cancel` seems to be completely unused, including testing.

Reply:
`stopCh` is not used either, I just replaced it with `context` and `cancel` in place and skipped considering its use. After all, if there is no `cancel`, it is also meaningless to just hold this `context`